### ISSUE TITLE
Reader._preprocess needs to call reshape before convert_dtype

### ISSUE THREAD

**Issue by lucasgreene** [vanhuyz/CycleGAN-TensorFlow#38]
Hi so in continuation with my earlier point about the conversion to [-1,1] not being correct in utils, i have uncovered a bit of an odd behavior in tensorflow that is relevant.

So the tf.image.resize_images function takes in an image and then resizes it using some interpolation function (default is bilinear). This **by default** will convert your image to float as it needs to interpolate scalars. 

The tf.image.convert_image_dtype function will convert your uint8 image in range [0,255] to the floating range [0,1], **but only if it is actually in uint8**. If you pass in an already floating point image with this call tf.image.convert_image_dtype(image, dtype=tf.float32) , the function **does **nothing**. 

So the current code is (in reader) 
`
  def _preprocess(self, image):
    image = tf.image.resize_images(image, size=(self.image_size, self.image_size))
    image = utils.convert2float(image)
    image.set_shape([self.image_size, self.image_size, 3])
    return image

`

The first part implicitly converts the image to float, which means the call to conver_image_dtype in utils.convert2float does nothing. This is why the current code sort of works, even though it is wrong. When i changed the /127.5 part in utils to *2, i got super high values and the order of the calls in preprocess is why.

In summary, 
  1) in my other issue I raised that you shouldn't divide by 127.5 in utils.conver2float rather you should multiply by 2
  2) after that is fixed the order of the image resizing and conversion needs to be reversed so you convert to float and then resize
